### PR TITLE
[SelectionManager] Use event.detail instead for counting the mousedown event

### PR DESCRIPTION
**Pull request by mofux**
This PR introduces the usage of MouseEvent.detail to get the number of consecutive clicks. 

See:
https://www.w3schools.com/jsref/event_detail.asp

This does not only reduce the complexity of the code base, but also respects the OS double click speed setting of the user.

### CODE REVIEW

**Comment by Tyriar** [comment]
This is pretty cool, I'm a little concerned about browser support though as I don't trust w3schools and mdn says `?` for most browsers https://developer.mozilla.org/en-US/docs/Web/API/UIEvent/detail

@rebornix you wouldn't happen to know why monaco doesn't use `UIEvent.detail` do you?

File: src/SelectionManager.ts [hunk 1]
```diff
@@ -396,16 +378,14 @@ export class SelectionManager extends EventEmitter {
...
       this._onShiftClick(event);
     } else {
-      if (this._clickCount === 1) {
+      if (event.detail === 1) {
           this._onSingleClick(event);
-      } else if (this._clickCount === 2) {
+      } else if (event.detail === 2) {
           this._onDoubleClick(event);
-      } else if (this._clickCount === 3) {
+      } else if (event.detail === 3) {
```
Comment (Tyriar) [inline src/SelectionManager.ts]: Should this be >= to maintain current behavior?

**Comment by mofux** [comment]
@Tyriar As far as I can tell, UIEvent.detail is well supported. I have tested it on Safari, Chrome, Firefox and IE10 a while back, all working. It is not very well documented, and I believe most people don't even know it exists. Actually, I have implemented a very similar solution to yours a few months ago, and then I've seen that cloud9 can handle the same thing, but with respecting the double click setting of the os, so I was wondering how they do it and found out that they were using evt.detail 😉

File: src/SelectionManager.ts (hunk 1 above)
Comment (mofux) [inline src/SelectionManager.ts]: I think you are right, more than three clicks should yield the same behaviour as three clicks.

File: src/SelectionManager.ts [hunk 2]
```diff
@@ -385,7 +385,7 @@ export class SelectionManager extends EventEmitter {
           this._onSingleClick(event);
       } else if (event.detail === 2) {
           this._onDoubleClick(event);
-      } else if (event.detail === 3) {
+      } else if (event.detail >= 3) {
```
Comment (Tyriar) [inline src/SelectionManager.ts]: Actually looking closer at the code it was correct before :sweat_smile:, we shouldn't need to re-perform triple click as we don't drop the selection between clicks 3 and 4.

**Comment by Tyriar** [comment]
Just revert that change and ready to go, pending the tests. I love deleting code :fireworks:

**Comment by mofux** [comment]
Ready for take-off 🚀

**Comment by Tyriar** [comment]
@parisk is it possible to make the mac lint/test Travis jobs non-block to merge? There is quite a queue for Mac builds apparently.

**Comment by parisk** [comment]
@Tyriar, yes of course.

**Pull request merged**